.github/workflows/periodic_benchmarks.yaml

```
@@ -0,0 +1,96 @@
+# Initial Source: pybop-team/PyBop
+
+# This workflow periodically runs the benchmarks suite in benchmarks/
+# using asv and publish the results, effectively updating
+# the display website hosted in the pybop-bench repo
+
+# Steps:
+# - Benchmark all commits since the last one that was benchmarked
+# - Push results to pybop-bench repo
+# - Publish website
+name: Benchmarks
+on:
+  # Everyday at 12 pm UTC
+  schedule:
+    - cron: "0 12 * * *"
+  # Make it possible to trigger the
+  # workflow manually
+  workflow_dispatch:
+
+jobs:
+  benchmarks:
+    runs-on: [self-hosted, macOS, ARM64]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install python & create virtualenv
+        shell: bash
+        run: |
+            eval "$(pyenv init -)"
+            pyenv install 3.11 -s
+            pyenv virtualenv 3.11 pybop-311-bench
```

```suggestion
            pyenv install 3.12 -s
            pyenv virtualenv 3.12 pybop-312-bench
```

i.e., thanks to the Faster CPython team, Python has been gaining speed in subsequent major releases since the past few years – should we try Python 3.12?